src/Models/ArduinoDeviceBase.ts

```diff
@@ -91,7 +98,25 @@ export abstract class ArduinoDeviceBase implements Device {
     if (!result) {
       return false;
     }
-    await vscode.commands.executeCommand('arduino.verify');
+
+    // Execute default compile task to compile device code.
+    const tasks = await vscode.tasks.fetchTasks();
+    if (!tasks || tasks.length < 1) {
+      const message = `Failed to fetch tasks.`;
+      utils.channelShowAndAppendLine(this.channel, message);
+      return false;
+    }
+
+    const arduinoCompileTask = tasks.filter(task => {
+      return task.name === constants.compileTaskName;
+    });
+    if (!arduinoCompileTask || arduinoCompileTask.length < 1) {

```

Comment:
what if there are two or more tasks for compileTask? should we also validate this situation?

Reply:
We select compile task by task name, which is defined in constants.compileTaskName.